examples/drivencavity.py

```diff
@@ -64,35 +145,41 @@ def main(nelems: int, etype: str, degree: int, reynolds: float):
 # results of Stokes and Navier-Stokes, and because of the extra steps required
 # for establishing streamlines.
 
+def postprocess(domain, ns, **arguments):
 
-def postprocess(domain, ns, every=.05, spacing=.01, **arguments):
-
-    ns = ns.copy_() # copy namespace so that we don't modify the calling argument
-    ns.add_field('stream', domain.basis('std', degree=2)[1:]) # stream function (sans first dof to obtain non-singular system)
-    ns.ε = function.levicivita(2)
-    sqr = domain.integral('Σ_i (u_i - ε_ij ∇_j(stream))^2 dV' @ ns, degree=4)
-    arguments = solver.optimize('stream,', sqr, arguments=arguments) # compute streamlines
+    # reconstruct velocity streamlines
+    sqr = domain.integral('Σ_i (u_i - ε_ij ∇_j(ψ))^2 dV' @ ns, degree=4)
+    arguments = solver.optimize('ψ,', sqr, arguments=arguments)
 
     bezier = domain.sample('bezier', 9)
-    x, u, p, stream = bezier.eval(['x_i', 'sqrt(u_i u_i)', 'p', 'stream'] @ ns, **arguments)
-    with export.mplfigure('flow.png') as fig:  # plot velocity as field, pressure as contours, streamlines as dashed
-        ax = fig.add_axes([.1, .1, .8, .8], yticks=[], aspect='equal')
-        import matplotlib.collections
-        ax.add_collection(matplotlib.collections.LineCollection(x[bezier.hull], colors='w', linewidths=.5, alpha=.2))
-        ax.tricontour(x[:, 0], x[:, 1], bezier.tri, stream, 16, colors='k', linestyles='dotted', linewidths=.5, zorder=9)
-        caxu = fig.add_axes([.1, .1, .03, .8], title='velocity')
-        imu = ax.tripcolor(x[:, 0], x[:, 1], bezier.tri, u, shading='gouraud', cmap='jet')
-        fig.colorbar(imu, cax=caxu)
-        caxu.yaxis.set_ticks_position('left')
-        caxp = fig.add_axes([.87, .1, .03, .8], title='pressure')
-        imp = ax.tricontour(x[:, 0], x[:, 1], bezier.tri, p, 16, cmap='gray', linestyles='solid')
-        fig.colorbar(imp, cax=caxp)
+    x, u, p, ψ = bezier.eval(['x_i', 'sqrt(u_i u_i)', 'p', 'ψ'] @ ns, **arguments)
+
+    with export.mplfigure('flow.png', dpi=150) as fig: # plot velocity as field, pressure as contours, streamlines as dashed
+        ax = export.triplot(fig, x, u, tri=bezier.tri, hull=bezier.hull, cmap='hot_r', clim=(0,1), vlabel='velocity')
+        ax.tricontour(x[:, 0], x[:, 1], bezier.tri, ψ, levels=numpy.sort(ψ)[::len(ψ)//32], colors='k', linestyles='solid', linewidths=.5, zorder=9)
```

Comment:
I'm not sure if it is an improvement, but you can also create levels using

```python
numpy.percentile(ψ, numpy.linspace(0, 100, 32))
```

Reply:
done